model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/role/JpaRoleModelCriteriaBuilder.java

```diff
@@ -61,6 +62,15 @@ public JpaRoleModelCriteriaBuilder compare(SearchableModelField<? super RoleMode
                     return new JpaRoleModelCriteriaBuilder((cb, root) -> 
                         cb.equal(root.get(modelField.getName()), value[0])
                     );
+                } else if (modelField == SearchableFields.COMPOSITE_ROLE){
```

Comment:
Actually, when thinking about it and reading[1], I believe we should be using `==` instead of `equals`. I'd create follow-up task to update it in all other places if we agree on it. I'll bring it up in the mtg.

[1] https://docs.oracle.com/javase/specs/jls/se9/html/jls-8.html#jls-8.9

Reply:
I agree with changing it to `==` as we want to have only one instance of each `SearchableField`, however, SearchableFields are not `enum`. That is probably why there was `equals` method used.

Reply:
+1 for using `==`, the fields should be the same instances as the predefined ones.

Reply:
https://github.com/keycloak/keycloak/issues/11843